From 4a8e9d80409590cb42732ae3105b5ae71fda52c1 Mon Sep 17 00:00:00 2001
From: Bin Liu <bliu@suse.com>
Date: Fri, 2 Dec 2016 10:37:27 +0800
Subject: [PATCH] Totempg: remove duplicate memcpy in mcast_msg func

In function mcast_msg of totempg.c, line 923, there is a memcpy call in
"else" branch, and also another memcpy out of the "else" branch, while
the two calls have the same parameters. It is possibleto remove the memcpy
in "else" branch.

Signed-off-by: Bin Liu <bliu@suse.com>
Reviewed-by: Jan Friesse <jfriesse@redhat.com>
(cherry picked from commit 819d66ca1ccaa86ef21fc4112033b38ce3f9af29)
---
 exec/totempg.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/exec/totempg.c b/exec/totempg.c
index 0b46782..05729be 100644
--- a/exec/totempg.c
+++ b/exec/totempg.c
@@ -934,8 +934,6 @@ static int mcast_msg (
 				data_ptr = (unsigned char *)iovec[i].iov_base + copy_base;
 			else {
 				data_ptr = fragmentation_data;
-				memcpy (&fragmentation_data[fragment_size],
-				(unsigned char *)iovec[i].iov_base + copy_base, copy_len);
 			}
 
 			memcpy (&fragmentation_data[fragment_size],
-- 
1.7.1

